### ISSUE TITLE
cmake build broken in 7.43.0

### ISSUE THREAD

**Issue by rleigh-dundee** [bagder/curl#326]
At least three `CMakeLists.txt` files are not included in your distribution tar files:

```
CMake Error at tests/CMakeLists.txt:1 (add_subdirectory):
  The source directory

    /tmp/curl-source/tests/data

  does not contain a CMakeLists.txt file.


CMake Error at tests/CMakeLists.txt:2 (add_subdirectory):
  The source directory

    /tmp/curl-source/tests/libtest

  does not contain a CMakeLists.txt file.


CMake Error at tests/CMakeLists.txt:3 (add_subdirectory):
  The source directory

    /tmp/curl-source/tests/server

  does not contain a CMakeLists.txt file.
```

Could you consider adding these and any other `CMakeLists.txt` files to `EXTRA_DIST` in Makefile.am or the system you use to generate releases if something different?  The files are present it git, but not the source release.  Thanks.

As a followup question, have you considered using `git archive` to distribute the tagged release, skipping automake's `make dist` entirely?  This means you always get what's in git, with no possibility of missing out any files, since you no longer need to explicitly include each one.  This also means the release sources are always exactly the same as the git commit for that release.

You might find these helpers of use:
  https://github.com/codelibre-net/schroot/blob/master/cmake/GitRelease.cmake [cmake]
  https://github.com/codelibre-net/schroot/blob/schroot-1.6/scripts/git-dist.mk [automake]
These replace `make dist` with targets to tag and sign the current release commit, and then generate a source release on a distribution branch, then finally you can run `git archive` to generate a ready-to-use source tarball directly from a signed tag.  i.e. it gives you a git release workflow.

Kind regards,
Roger

**Comment by bagder** [comment]
The sources in the release archives are actually intentionally not exactly the same as in git, as we generate a few of them at release. Also, it is handy to allow more files in git than in the release archives.

**Comment by bagder** [comment]
I merged #327 just now so this problem is fixed.
